runtime/src/main/java/org/corfudb/runtime/collections/SMRMultiIndex.java

```diff
@@ -0,0 +1,374 @@
+            return Collections.EMPTY_SET;
+        Map<K,V> index = indexMap.get(indexKey);
+        if (index == null)
+            return Collections.EMPTY_SET;
+        List result = new ArrayList();
+        for (Map.Entry<K,V> entry : index.entrySet()) {
+            result.add(projectionFunction.generateProjection(entry.getKey(), entry.getValue(), indexKey));
+        }
+        return result;
+    }
+
+    /** Get a value by a given named index.
+     *
+     * @param indexName  The index of the index function from the list
+     *                  passed in at construction time.
+     * @param indexKey The index Key to use.
+     * @return          A set of values which match the given index criteria.
+     */
+    @Accessor
+    public Collection<Map.Entry<K,V>> getByNamedIndexUnsafe(String indexName, I indexKey) {
+        Map<I,Map<K,V>> indexMap = indexMaps.get(indexName);
+        if (indexMap == null)
+            return Collections.EMPTY_SET;
+        Map<K,V> index = indexMap.get(indexKey);
+        if (index == null)
+            return Collections.EMPTY_SET;
+        return index.entrySet();
+    }
+
+    /** Get the size of the multi-index.
+     * @return  The number of entries mapped (rows).
+     */
+    @Accessor
+    public int size() {
+        return mainMap.size();
+    }
+
+    @Accessor
+    @Override
+    public boolean isEmpty() {
+        return mainMap.isEmpty();
+    }
+
+    @Accessor
+    @Override
+    public boolean containsKey(Object key) {
+        return mainMap.containsKey(key);
+    }
+
+    @Accessor
+    @Override
+    public boolean containsValue(Object value) {
+        return mainMap.containsValue(value);
+    }
+
+
+
+    /** Clear the multi-index.
+     */
+    @Mutator(reset = true, name = "clear")
+    public void clear() {
+        throw new RuntimeException("###########Operation Not Implemented.##########");
+
+    }
+
+    @Accessor
+    @Override
+    public Set<K> keySet() {
+        return mainMap.keySet();
+    }
+
+    @Accessor
+    @Override
+    public Collection<V> values() {
+        return mainMap.values();
+    }
+
+    @Accessor
+    @Override
+    public Set<Entry<K, V>> entrySet() {
+        return mainMap.entrySet();
+    }
+
+
+    //--------------- Helper Methods to Create/Update/Remove Indexes -----------
+
+    public void createIndexes(K key, V value) {
+        indexSpecifications.entrySet().forEach(entry -> {
+            getIndexList(entry.getKey(), entry.getValue(), key, value).forEach(index -> index.put(key, value));
+        });
+    }
+    public void removeIndexes(K key, V value) {
+        indexSpecifications.entrySet().forEach(entry -> {
+            getIndexList(entry.getKey(), entry.getValue(), key, value).forEach(index -> index.remove(key));
+        });
+    }
+
+    public List<Map<K, V>> getIndexList(String indexName, IndexSpecification indexSpecification, K key, V value) {
+
+        List<Map<K, V>> indexes = new ArrayList<Map<K, V>>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)